### ISSUE TITLE
Update pereceptron.ipynb

### ISSUE THREAD

**Issue by leestott** [microsoft/AI-For-Beginners#553]
### Objective
Modify the `Perceptron` class in the notebook to include a configurable learning rate and apply it during weight updates, aligning the implementation with gradient descent principles.

#### 1. **Locate the Perceptron Class**
- File: `Perceptron.ipynb`
- Section: Where the `Perceptron` class is defined

#### 2. **Update the Constructor**
- Add a `learning_rate` parameter to `__init__` with a default value (e.g., `0.01`)
- Store it as an instance variable: `self.learning_rate`

```python
def __init__(self, input_size, learning_rate=0.01):
    self.weights = np.zeros(input_size)
    self.learning_rate = learning_rate
```

#### 3. **Modify the Weight Update Logic**
- In the `train` method, update the weights using the learning rate:

```python
self.weights += self.learning_rate * error * x
```

#### 4. **Update Any Instantiations of Perceptron**
- Wherever `Perceptron` is instantiated, optionally pass a learning rate:

```python
model = Perceptron(input_size=2, learning_rate=0.01)
```

#### 5. **Add a Markdown Cell Explaining the Change**
- Briefly explain what a learning rate is and why it matters in gradient descent
- Mention that the code now reflects this concept

#### 6. **Test the Updated Notebook**
- Run all cells to ensure the perceptron still trains correctly
- Confirm that learning rate affects convergence speed

#### 7. **Commit Changes**
- Use a clear commit message:
  ```
  Added learning rate to Perceptron class and updated training logic
  ```

###  Enhancements
- Add a slider widget to experiment with different learning rates interactively
- Plot training accuracy vs. learning rate to visualize its impact

### Validate 
Ensure the gradient descent formula shown  accurately represent the perceptron update rule being implemented. Consider updating the formula to match the implementation: w^(τ+1) = w^τ + η * x_n (for misclassified positive) or w^(τ+1) = w^τ - η * x_n (for misclassified negative).

#### 8. **Push and Create Pull Request**
- Title: `Add learning rate to Perceptron training`
- Description: Explain the motivation and what was changed
- Tag reviewers if needed
